### DISCUSSION TITLE
Detect connections

### DISCUSSION THREAD

**Discussion by imfelipedev** [vercel/next.js#56849] [Help]
### Summary

How can I detect when a user logs into my website and leaves. At application level (Next 13, app).

**Comment by icyJoseph** [comment]
Use a visibility change event, https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event:

> This event fires with a visibilityState of `hidden` when a user navigates to a new page, switches tabs, closes the tab, minimizes or closes the browser, or, on mobile, switches from the browser to a different app. Transitioning to `hidden` is the last event that's reliably observable by the page, so developers should treat it as the likely end of the user's session (for example, for [sending analytics data](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon)).

You can put a Client Component as a child of the RootLayout of your app, and that one can attach this event listener. 

You might event measure session length and such, and see if the user is coming back to the tab.

**Reply by imfelipedev** [reply]
I got it, thank you very much

**Reply by icyJoseph** [reply]
Awesome. Please don't forget to also mark as answered.
